python/src/helpers/count_matrix.py

```diff
@@ -22,84 +27,57 @@ def get_objects(self):
         if not objects:
             return {}
 
-        objects = {o["Key"]: o["ETag"] for o in objects if o["Size"] > 0}
+        objects = {o["Key"]: o["LastModified"] for o in objects if o["Size"] > 0}
 
         return objects
 
-    def md5_full_file(self, file_path):
-        hash_md5 = hashlib.md5()
-        with open(file_path, "rb") as f:
-            for chunk in iter(lambda: f.read(4096), b""):
-                hash_md5.update(chunk)
-        return hash_md5.hexdigest()
-
-    def megabyte_chunks(self, filesize, num_parts):
-        x = filesize / int(num_parts)
-        y = x % 1048576
-        return int(x + 1048576 - y)
-
-    def validate_etag(self, file_path, etag):
-
-        # get the size of the file. if it doesn't exist,
-        # the etags definitely don't match
-        try:
-            filesize = os.path.getsize(file_path)
-        except OSError:
-            return False
+    @xray_recorder.capture("CountMatrix.download_object")
```

Comment:
What information is saved by the recorder when you use capture for a function?

Reply:
It adds an extra event to the trace so we know what function the request was in in that given time. It creates a subsegment with the scope of the function. See, e.g. in the picture, where a subsegment is `WorkSubmitService.getQueueAndHandleMessage` and the AWS call is scoped under it.

![image](https://user-images.githubusercontent.com/16019444/113326676-e6bff900-9311-11eb-8cb3-e29c2e5a06d3.png)